.github/workflows/dev_deployment.yml

```diff
@@ -0,0 +1,35 @@
+name: Initiate Dev Environmnet Deployment
+
+on:
+    push: #Add back master branch after testing
+env:
+    DEV_VM_IP: '${{secrets.DEV_VM_IP}}'
+    DEV_VM_PSWD: '${{secrets.DEV_VM_PSWD}}'
+    DISCORD_WH: '${{secrets.DISCORD_WEBHOOK}}'
+jobs:
+    deploy:
+        runs-on: ubuntu-latest
+        strategy:
+            matrix:
+                python-version: ["3.10"]
+
+        steps:
+            - uses: actions/checkout@v3
+
+            - name: Set up Python ${{ matrix.python-version }}
+              uses: actions/setup-python@v4
+              with:
+                python-version: ${{ matrix.python-version }}
+
+            - name: Install dependencies
+              run: |
+                python3 -m pip install --upgrade pip
+                pip3 install paramiko
+
+            - name: Extract branch name
+              id: extract_branch_name
+              run: echo "branch=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}" >> $GITHUB_OUTPUT
+
+            - name: Start deployment
+              run: |
+                ./start_deployment.py --ip ${DEV_VM_IP} --pswd ${DEV_VM_PSWD} --discordwh ${DISCORD_WH} --branch ${{ steps.extract_branch_name.outputs.branch }}
```

Comment:
Your implementation is not working assuming you've run the script:
<img width="446" alt="image" src="https://github.com/AUBGTheHUB/monolith/assets/104720011/57e10ef1-18c0-4efe-98c5-a5c58121b282">

Reply:
The branch should've been `deployment-workflows`